index.bs

```diff
@@ -3133,6 +3143,116 @@ The [=remote end steps=] with |command parameters| are:
 
 </div>
 
+#### The browsingContext.traverseHistory Command ####  {#command-browsingContext-traverseHistory}
+
+The <dfn export for=commands>browsingContext.traverseHistory</dfn> command
+traverses the history of a given navigable by a delta.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      BrowsingContext.TraverseHistory = {
+        method: "browsingContext.traverseHistory",
+        params: browsingContext.TraverseHistoryParameters
+      }
+
+      browsingContext.TraverseHistoryParameters = {
+        context: browsingContext.BrowsingContext,
+        delta: js-int,
+      }
+      </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+        browsingContext.TraverseHistoryResult = {
+        }
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for browsingContext.traverseHistory">
+The [=remote end steps=] with |command parameters| are:
+
+1. Let |browsing context| be the result of [=trying=] to [=get a browsing context=]
+   with |command parameters|["<code>context</code>"].
+
+1. Assert: |browsing context| is not null.
+
+1. Let |navigable| be the [=/navigable=] whose [=navigable/active
+   document=] is |browsing context|'s [=browsing context/active document=].
+
+1. Let |delta| be |command parameters|["<code>delta</code>"].
+
+1. Let |resume id| be a unique string.
+
+1. [=Queue a task=] on |navigable|'s [=session history traversal queue=] to run
+   the following steps:
+
+    1. Let |all steps| be the result of [=getting all used history steps=] for |navigable|.
+
+    1. Let |current index| be the index of |navigable|'s [=current session history
+       step=] within |all steps|.
+
+    1. Let |target index| be |current index| plus |delta|.
+
+    1. Let |valid entry| be false if |all steps|[|target index|] does not exist,
```

Comment:
Won’t this always be false?

Reply:
Why? If I open a new tab and then try to navigate forward then `|all steps|` will have length 1, `|current index|` will be `0` and `|target index|` will be `1`, so `|all steps|[|target index|]` won't exist (also I copied the logic from HTML, so I'm fairly sure it's right).

Reply:
Oh, I see, the context didn't show me the incorrect line. Sorry!